File: JavaReactKampHomeWork/homeworkJavaSE1.8June/src/day4Homework3GamerBackEnd/abstracts/BaseGameManager.java
package day4Homework3GamerBackEnd.abstracts;




import day4Homework3GamerBackEnd.entities.Game;

public abstract class BaseGameManager implements GameSalesService,GameService{
	
	@Override
	public void orderInBulk(Game game) {
	int value = game.getOrderAmount();
	boolean isNewOrder = false;
	if(game.getAmountInStock()<7 ) {
		game.setOrderAmount(12);
    	isNewOrder = true;	
    	value += game.getOrderAmount();
	    System.out.println(game.getGameTitle() + " oyunu stoklarimizda gecici sure ile bulunmamaktadir.");
			    	
		    	
	}
	System.out.println("Yeni siparis olacak mi : (" + isNewOrder + ") " + value);
				
	}
	@Override
	public void addDiscount(Game game) {
		System.out.println("Indirimli Fiyat (" + game.getGameTitle() +"): " + game.getPriceAfterDiscount());
	}

	@Override
	public void giveGifts(Game game) {
		System.out.println("Bu alisveristen kazandigin puan tutari : " + game.getMarketValue()*game.getGiftAmount()/100);				
	}

	@Override
	public void updateStock(Game game) {
		System.out.println("Su anki stok (" + game.getGameTitle() + ") : " +game.getAmountInStock());
		game.setAmountInStock(game.getAmountInStock() + game.getOrderAmount());
		System.out.println("Forecasted (" + game.getGameTitle() + ") : " +game.getAmountInStock());
	}
	@Override
	public void addNew(Game game) {
		System.out.println(game.getGameTitle() + " eklendi");
	}
	
	

	

}

File: YazilimGelistiriciDotNetHomeWork/source/repos/KampHomeWorkJune/Day5Homework5GamerBackEnd/Abstracts/BaseGameManager.cs
using Day5Homework5GamerBackEnd.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace Day5Homework5GamerBackEnd.Abstracts
{
    public abstract class BaseGameManager : IGameService, IGameSalesService
    {
        public void AddDiscount(Game game)
        {

            Console.WriteLine( "Indirimli Fiyat (" + game.GameTitle + "): " + game.PriceAfterDiscount);
        }

        public void AddNew(Game game)
        {
            Console.WriteLine(game.GameTitle + " eklendi");
        }

        public void GiveGifts(Game game)
        {
            Console.WriteLine("Bu alisveristen kazandigin puan tutari : " + game.MarketValue * game.GiftAmount / 100);
        }

        public void OrderInBulk(Game game)
        {
            int value = game.OrderAmount;
            bool isNewOrder = false;
            if (game.AmountInStock < 7)
            {
                game.OrderAmount = 12;
                isNewOrder = true;
                value += game.OrderAmount;
                Console.WriteLine(game.GameTitle + " oyunu stoklarimizda gecici sure ile bulunmamaktadir.");
            }
            Console.WriteLine("Yeni siparis olacak mi : (" + isNewOrder + ") " + value);
        }
        public void UpdateStock(Game game)
        {
            Console.WriteLine ("Su anki stok (" + game.GameTitle + ") : " + game.AmountInStock);
            game.AmountInStock = game.AmountInStock + game.OrderAmount;
            Console.WriteLine("Forecasted (" + game.GameTitle + ") : " + game.AmountInStock);
        }
    }
}
